src/algebra/ordered_group.lean

```diff
@@ -831,13 +833,24 @@ lemma abs_abs_sub_abs_le_abs_sub (a b : α) : abs (abs a - abs b) ≤ abs (a - b
 abs_sub_le_iff.2 ⟨abs_sub_abs_le_abs_sub _ _, by rw abs_sub; apply abs_sub_abs_le_abs_sub⟩
 
 lemma abs_eq (hb : 0 ≤ b) : abs a = b ↔ a = b ∨ a = -b :=
```

Comment:
is the `mp` of this lemma worth it, without the need for `hb`?

Reply:
I had thought that already exists or is true via simple chain of definitions but upon further investigation I think I may have been misremembering. So perhaps it is worth adding that version, in which case you can add
```
lemma abs_eq (h : abs a = b) : a = b ∨ a = -b := 
by simpa only [← h, eq_comm, eq_neg_iff_eq_neg] using abs_choice a
```
and your proof `abs_eq` would become
```
lemma abs_eq_iff (hb : 0 ≤ b) : abs a = b ↔ a = b ∨ a = -b :=
begin
  refine ⟨λ h, abs_eq h, _⟩,
  rintro (h | h); simp only [h, abs_neg, abs_of_nonneg hb],
end
```

Reply:
Actually, the name of the first lemma should probably be something like `eq_of_abs_eq`.

Reply:
If you can come up with a good name :)

Reply:
How's `eq_or_eq_neg_of_abs_eq`? And then the other two lemmas can be `abs_eq` and `abs_eq_abs` (the other lemmas in this section, e.g. `abs_le`, don't have `iff` in their names).

Reply:
Apparently github showed me an outdated page and I didn't see your comments. Changed to `eq_or_eq_neg_of_abs_eq` anyway.